clang/utils/TableGen/ClangAttrEmitter.cpp

```diff
@@ -49,40 +49,38 @@ using namespace llvm;
 namespace {
 
 class FlattenedSpelling {
-  std::string V, N, NS;
+  StringRef V, N, NS;
```

Comment:
I'm not keen on this change; it may be correct, but it's pretty dangerous too because this now requires `FlattenedSpelling` to outlive the memory owned by the passed in string references. Keeping a `std::string` here introduces a copy, but that seems reasonable (tablegen is not really performance critical anyway). WDYT?

Reply:
In all these instances, the strings are derived from TableGen Records, so they do outlive. I do agree that TG is not perf critical, so if we prefer peace of mind here, we can just go with my previous PR. On the flip side, so many other TableGen backends rely on this behavior (that is, StringRefs derived from Records are stashed in other places, for example see `CodeGenIntrinsic` so this is in keeping with the current precedent).

Reply:
>  TG is not perf critical

I have seen bugs about Tg perf, but may be not this particular backend,

Reply:
In short, I'll defer it to you. If we don't want to commit this, I am fine and the find the reasoning behind if reasonable as well :)